composer/trainer/checkpoint.py

```diff
@@ -25,14 +26,29 @@ class CheckpointLoader:
         checkpoint_filepath (str): The path to an existing checkpoint file.
     """
 
-    def __init__(self, checkpoint_filepath: str):
+    def __init__(self, checkpoint_filepath: str, load_weights_only: bool = False, strict: bool = False):
         self.state_dict = torch.load(checkpoint_filepath, map_location='cpu')
+        self.load_weights_only = load_weights_only
+        self.strict = strict
```

Comment:
How would the strict parameter be used if `load_weights_only is False`?

Reply:
I added validation, so `strict` cannot be used if `load_weights_only = False`